code/entities/weapons/fnb/GooglyEye.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Sandbox;
+
+namespace Cinema;
+
+public partial class GooglyEye : WeaponBase
+{
+    static private float MaxPlacementDistance = 96.0f;
+    PreviewEntity PreviewModel { get; set; }
+
+    public override void ClientSpawn()
+    {
+        base.ClientSpawn();
+
+        if (!PreviewModel.IsValid()) 
```

Comment:
Perhaps we can change this into:

PreviewModel ?= new PreviewEntity

Just cause its clean

Reply:
I moved this to be created/deleted on Active Start/End. I couldn't get the ??= operator to work, but I did simplify it